src/components/implementation/fork_tests/voter/voter.c

```diff
@@ -89,17 +92,20 @@ int nwrite(spdid_t spdid, replica_type to, size_t sz) {
 		return -1; // whatever, shouldn't happen if blocked
 	}
 	else {
+		assert(replica->buf_write);
+		assert(sz <= 1024);
+		
 		replica->destination = to_comp;
+		replica->sz_write = sz;
 		printc("resplica destination set to %x\n", replica->destination);
-		assert(replica->buf_write);
 		ret = block_replica(replica);
 		if (ret < 0) printc("Error\n");
 		/* On wakeup */	
 		return 0;
 	}
 }
 
-int nread(spdid_t spdid, replica_type from, size_t sz) {
+size_t nread(spdid_t spdid, replica_type from, size_t sz) {
```

Comment:
`replica_type` is not correct for two reasons.

First, if you're going to provide a new type (i.e. with `typedef`), it should be `replica_t`.

Second, we are supposed to be reading and writing to channels, not specific replicas.  Why would the component being replicated even know it is replicated?  Ideally, we want to hide that from it.  There should be no hint about replicas anywhere in the API.

We read and write to *channels*, so that variables should reflect that.